Libraries/LibWeb/TrustedTypes/TrustedTypePolicyFactory.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, Miguel Sacristán Izcue <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/TrustedTypes/TrustedTypePolicyFactory.h>
+
+#include <LibGC/Ptr.h>
+#include <LibJS/Runtime/Realm.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/DOM/QualifiedName.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/AttributeNames.h>
+#include <LibWeb/HTML/GlobalEventHandlers.h>
+#include <LibWeb/HTML/WindowEventHandlers.h>
+#include <LibWeb/Namespace.h>
+
+namespace Web::TrustedTypes {
+
+GC_DEFINE_ALLOCATOR(TrustedTypePolicyFactory);
+
+GC::Ref<TrustedTypePolicyFactory> TrustedTypePolicyFactory::create(JS::Realm& realm)
+{
+    return realm.create<TrustedTypePolicyFactory>(realm);
+}
+
+Optional<String> TrustedTypePolicyFactory::get_attribute_type(String& tagName, String& attribute, Optional<String> elementNS, Optional<String> attrNS)
+{
+    // 1. Set localName to tagName in ASCII lowercase.
+    auto const localName = tagName.to_ascii_lowercase();
+
+    // 2. Set attribute to attribute in ASCII lowercase.
+    attribute = attribute.to_ascii_lowercase();
+
+    // 3. If elementNs is null or an empty string, set elementNs to HTML namespace.
+    if (!elementNS.has_value() || elementNS.value().is_empty())
+        elementNS = String { Namespace::HTML };
+
+    // 4. If attrNs is an empty string, set attrNs to null.
+    if (attrNS.has_value() && attrNS.value().is_empty())
+        attrNS.clear();
+
+    // FIXME
+    // 5. Let interface be the element interface for localName and elementNs.
+    auto const& interface = localName;
+
+    // 6. Let expectedType be null.
+    Optional<String> expectedType {};
+
+    // 7. Set attributeData to the result of Get Trusted Type data for attribute algorithm, with the following arguments, interface as element, attribute, attrNs
+    auto const attributeData = get_trusted_type_data_for_attribute(interface, attribute, attrNS);
+
+    // 8. If attributeData is not null, then set expectedType to the interface’s name of the value of the fourth member of attributeData.
+    if (attributeData.has_value()) {
+        expectedType = attributeData.value().trustedType;
+    }
+
+    // 9. Return expectedType.
+    return expectedType;
+}
+
+TrustedTypePolicyFactory::TrustedTypePolicyFactory(JS::Realm& realm)
+    : PlatformObject(realm)
+{
+}
+
+void TrustedTypePolicyFactory::initialize(JS::Realm& realm)
+{
+    WEB_SET_PROTOTYPE_FOR_INTERFACE(TrustedTypePolicyFactory);
+    Base::initialize(realm);
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#abstract-opdef-get-trusted-type-data-for-attribute
+Optional<TrustedTypeData> get_trusted_type_data_for_attribute(String const& element, String const& attribute, Optional<String> const& attributeNs)
+{
+    // 1.Let data be null.
```

Comment:
I know it makes no functional difference, but it would be closer to the spec to define `data` as an empty optional here and return it at step 4.

nit: missing space.

Reply:
The compiler will output the exact same thing so lets keep it close to the spec :)